### DISCUSSION TITLE
Troubleshooting Help

### DISCUSSION THREAD

**Discussion by DoragoDoggo** [dreamworksanimation/openmoonray#55] [Q&A] [answered]
Hi, I'm trying to build Moonray on Windows using a Docker container, but I keep running into two issues:
the first is that when I do `docker run -v E:/openmoonray:/openmoonray:shared -v /tmp:/tmp:shared --network=host --rm -it openmoonray_build`, it tells me "docker: Error response from daemon: path /tmp is mounted on / but it is not a shared mount."

I can run the command without mounting tmp, but if I do _that_ then when I try `cmake --preset container-release` I get this error:  "CMake Error at arras/arras_render/CMakeLists.txt:38 (find_package):
  By not providing "FindOpenImageIO.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "OpenImageIO", but CMake did not find one.

  Could not find a package configuration file provided by "OpenImageIO" with
  any of the following names:

    OpenImageIOConfig.cmake
    openimageio-config.cmake

  Add the installation prefix of "OpenImageIO" to CMAKE_PREFIX_PATH or set
  "OpenImageIO_DIR" to a directory containing one of the above files.  If
  "OpenImageIO" provides a separate development package or SDK, be sure it
  has been installed."
  
In another thread I noticed people having trouble downloading all the submodules, so I'm wondering if this is the case here, too. I already had to manually download some of the submodules because when I do `git clone --recurse-submodules https://github.com/dreamworksanimation/openmoonray.git` it tells me this at the end:
"error: invalid path 'houdini/otls/Vop::DW_MOONRAY::AttributeMap::1.hda'
fatal: Unable to checkout 'ce861215cf2c4fec77c8decdabd96755917ae742' in submodule path 'moonray/moonray_dcc_plugins'" That and all the following submodules within that folder fail to download aside from a .git file.

I could really use some help figuring this out.

**Comment by ereiames** [comment]
You were using

docker run -v E:/openmoonray:/openmoonray:shared -v /tmp:/tmp:shared --network=host --rm -it openmoonray_build

But /tmp does not make sense on a windows machine.

Try creating the directory E:/tmp on your windows machine and then use the following slightly modified command

docker run -v E:/openmoonray:/openmoonray:shared -v E:/tmp:/tmp:shared --network=host --rm -it openmoonray_build

and that should work for you (if the prior steps worked properly)

**Reply by DoragoDoggo** [reply]
Well, that took care of the first part of my issue, thank you! Now I just need to understand what's causing the other problem!

**Reply by DoragoDoggo** [reply]
Well, I figured out the error with the OpenImageIO thing. I dunno what I did, but it's working now, and it said "Configuring done" when running the `cmake --preset container-release ` command. But now when I do `cmake --build --preset container-release -- -j 64` it says "gmake: *** [Makefile:146: all] Error 2". Have I mentioned that I'm not a programmer?

**Reply by DoragoDoggo** [reply]
It also says "In file included from /openmoonray/moonray/moonray_arras/mcrt_computation/lib/common/rec_load/RecLoad.cc:6:
/openmoonray/moonray/moonray_arras/mcrt_computation/lib/common/rec_load/RecLoad.h:16:10: fatal error: scene_rdl2/common/rec_time/RecTime.h: No such file or directory
   16 | #include <scene_rdl2/common/rec_time/RecTime.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [moonray/moonray_arras/mcrt_computation/lib/common/rec_load/CMakeFiles/common_rec_load.dir/build.make:76: moonray/moonray_arras/mcrt_computation/lib/common/rec_load/CMakeFiles/common_rec_load.dir/RecLoad.cc.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:20981: moonray/moonray_arras/mcrt_computation/lib/common/rec_load/CMakeFiles/common_rec_load.dir/all] Error 2"

**Comment by ereiames** [comment]
I’m stuck at this error as well. I’ve had to step away and hope to get back to it in a few days. I could try to update the cmake setup but that feels wrong. Sorry if you are on a shorter timeline.Sent from iSomewhereOn Mar 20, 2023, at 23:03, DoragoDoggo ***@***.***> wrote:﻿
It also says "In file included from /openmoonray/moonray/moonray_arras/mcrt_computation/lib/common/rec_load/RecLoad.cc:6:
/openmoonray/moonray/moonray_arras/mcrt_computation/lib/common/rec_load/RecLoad.h:16:10: fatal error: scene_rdl2/common/rec_time/RecTime.h: No such file or directory
16 | #include <scene_rdl2/common/rec_time/RecTime.h>
|          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
gmake[2]: *** [moonray/moonray_arras/mcrt_computation/lib/common/rec_load/CMakeFiles/common_rec_load.dir/build.make:76: moonray/moonray_arras/mcrt_computation/lib/common/rec_load/CMakeFiles/common_rec_load.dir/RecLoad.cc.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:20981: moonray/moonray_arras/mcrt_computation/lib/common/rec_load/CMakeFiles/common_rec_load.dir/all] Error 2"

**Reply by DoragoDoggo** [reply]
My only timetable is my intense desire to just get this to work. My stubbornness prevents me from giving up, and this adversity only makes me more determined.

**Comment by evanspree3d** [comment, marked as answer]
I am through this and have it building/rendering. I cleared away everything and did that initial clone from a wsl2 shell (not PowerShell as I had before). Something someone wrote in one of these posts suggested this to me. Something downloads slightly differently when done from linux/unix than when done from windows. By downloading in the wsl2 shell and placing in the same preferred place on windows, everything else just worked.

Specifically, instead of just doing 

cd c:\users\username\documents\dev
git clone --recurse-submodules https://github.com/dreamworksanimation/openmoonray.git

in PowerShell or CMD, I opened a wsl2 shell (I think mine is Ubuntu but it likely does not matter) and there cd-ed to the windows location of my choosing (where "username" is just your windows username) and in this location (on windows) where I wanted clone this in the first place (I did delete the old openmoonray folder first):

cd /mnt/c/users/username/documents/dev
git clone --recurse-submodules https://github.com/dreamworksanimation/openmoonray.git

and from there on, everything worked exactly as advertised. Hope that helps (that is the only thing I needed the wsl2 shell for).

**Reply by DoragoDoggo** [reply]
This worked, thank you! Downloaded Windows Terminal, then opened a new window that's a WSL2 shell, which comes in the form of Ubuntu if you ran the `wsl --install` command, and everything worked after that.

Only issue I had is the build at the `cmake --build --preset container-release -- -j 64` step kept stopping and glitching out and I had to restart Docker to make it work again, but I think that's probably just something weird going on with my computer specifically.

But now I was able to do the test render and everything.
